tools/executable_analysis_tools/elf-float-checker.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Script for checking for floating point symbols in ELF files."""
+
+import argparse
+import logging
+import re
+import subprocess
+import sys
+
+log = logging.getLogger(__name__)
+
+LOG_FORMAT = "%(asctime)s - %(levelname)s - %(message)s"
+FLOATING_POINT_SYMBOL_REGEX = r"__aeabi_(cd.+|cf.+|h2f.+|d.+|f.+|.+2d|.+2f)"
```

Comment:
this regex is too broad as it appears to catch non floating point symbols as well

Reply:
Can you give an example of what was caught that was not a floating point symbol?

Reply:
my mistake, they are all indeed fp symbols